### PR TITLE
name changes

### PR DESCRIPTION
**Pull request by apmilen**
@nmushegian please review

### CODE REVIEW

File: src/math.sol [hunk 1]
```diff
@@ -16,10 +16,26 @@ contract DSMath {
         assert((z = x + y) >= x);
     }
 
+    function add(uint128 x, uint128 y) constant returns (uint128) {
+        return incr(uint128(x), uint128(y));
```
Comment (nmushegian) [inline src/math.sol]: redundant casts

**Comment by nmushegian** [comment]
lgtm (based on just the diff)

**Comment by nmushegian** [comment]
shoot we have to make tests dont we

**Comment by dbrock** [comment]
Can we please put these in a separate package `ds-math-long-names`?

**Comment by dbrock** [comment]
Another possible solution would be the following:

* Rename `incr` and `decr` to `add` and `sub`
* Add new functions `mul` and `div` which simply multiply and divide `uint128` integers (_not_ fixed-point numbers) with overflow checking

So we would have

* `add(uint128, uint128) returns (uint128)`
* `sub(uint128, uint128) returns (uint128)`
* `mul(uint128, uint128) returns (uint128)`
* `div(uint128, uint128) returns (uint128)`

which do the obvious thing ("normal" addition, subtraction, multiplication and division).

And then we simply keep `wmul`, `wdiv`, `rmul`, `rdiv` and `rpow` as-is.

**Comment by dbrock** [comment]
Another issue however is that these are all using the `uint128` type... which we ourselves have decided is the largest numerical type that we ever want to use, so this works great for us. But as @apmilen have requested maybe it would be nice to have `uint256` versions as well. Maybe even `int256`?

**Comment by dbrock** [comment]
Also the current `min` and `max` functions would then become `hmin` and `hmax`, and we would add new functions `min`, `max`, `imin` and `imax` that operate on `uint256` and `int256` numbers, respectively.

(And we could also of course add `wmin`, `wmax`, `rmin`, `rmax` aliases for completeness.)

**Comment by dbrock** [comment]
Under this paradigm if you are a normal, honest, hard-working human being and only use `uint256` (or `uint`, as it's usually written) you would only have to care about `add`, `sub`, `mul`, `div`, `min`, `max`.

Only if you are an advanced user who needs to use either wads, rays, or signed integers would you have to learn about the admittedly somewhat scary-looking prefixed functions.

Doesn't that seem like a good compromise?

**Comment by dbrock** [comment]
If we do this, I wouldn't even be against adding the following aliases:

* `subtract` := `sub`
* `multiply` := `mul`
* `divide` := `div`
* `raise` := `pow` ??

Or alternatively, which would probably be my preference:

* `plus` := `add`
* `minus` := `sub`
* `times` := `mul`
* `dividedBy` := `div`
* `raisedTo` := `pow`

I wouldn't really be for it either, but at least those are a manageable subset. (I really doubt if anybody would ever choose to use `wadMultiply` or `rayPow` over `wmul` and `rpow`.)

**Comment by GriffGreen** [comment]
I prefer the redundant checks that openZepplin performs on their multiply and divide functions.

function mul(uint a, uint b) internal returns (uint) {
    uint c = a * b;
    assert(a == 0 || c / a == b);
    return c;
  }

  function div(uint a, uint b) internal returns (uint) {
    assert(b > 0);
    uint c = a / b;
    assert(a == b * c + a % b);
    return c;
  }

**Pull request closed without merging**